signac/synced_collections/data_types/attr_dict.py

```diff
@@ -49,13 +49,17 @@ def __setattr__(self, key, value):
         # the object has been fully instantiated. We may want to add a try
         # except in the else clause in case someone subclasses these and tries
         # to use d['foo'] inside a constructor prior to _data being defined.
-        if key.startswith("__") or key in self._PROTECTED_KEYS:
+        # The order of these checks assumes that setting protected keys will be
+        # much more common than setting dunder attributes.
+        if key in self._PROTECTED_KEYS or key.startswith("__"):
             super().__setattr__(key, value)
         else:
             self.__setitem__(key, value)
 
     def __delattr__(self, key):
-        if key.startswith("__") or key in self._PROTECTED_KEYS:
+        # The order of these checks assumes that setting protected keys will be
+        # much more common than setting dunder attributes.
```

Comment:
This whole comment is kind of irrelevant because we don't expect deletion of protected keys or dunder attributes to be allowed... but whatever.
```suggestion
        # The order of these checks assumes that deleting protected keys will be
        # much more common than deleting dunder attributes.
```

Reply:
Yeah I mean it's possible that some internal method at some point will want do `del self._data` or something temporarily 🤷 .